src/events/ready.rs

```diff
@@ -54,3 +60,91 @@ pub async fn load_muted_members(ctx: &serenity::Context, _: &serenity::Ready) {
         }
     }
 }
+
+/// Creates/updates the role embed and starts collecting interactions.
+pub async fn prepare_role_embed(ctx: serenity::Context) -> Result<(), serenity::Error> {
+    let data = get_data_lock(&ctx).await;
+    let data = &mut *data.write().await;
+
+    update_role_embed(&ctx, data).await?;
+
+    tokio::task::spawn(async move {
```

Comment:
Why not just listen on interaction events in the event handler?

Reply:
Originally, doing it like this would have allowed me to put basically all the relevant code for the feature in one place, but seeing as I extracted the embed update/creation to its own function in utils to support `/reload`, that does not really apply anymore.
It would probably be cleaner to move it to the event handler then.